README.rst

```diff
@@ -454,16 +454,184 @@ template ``context_processors`` in ``settings.py`` as shown below.
++------------------+--------------------------------------------------------------+
+| ``url``          | (``str``) url of the link.  (``required``)                   |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the link icon.  (``optional``)   |
++------------------+--------------------------------------------------------------+
+
+Create a model link
+^^^^^^^^^^^^^^^^^^^
+
+If you want to create a link that contains a url of ``add`` or ``list`` page of a model
+then you can use following syntax. Users will only be able to see links for 
+models they have permission to either view or edit.
+
+**Syntax:**
+
+.. code-block:: python
+
+    # use to create a link of list page
+    my_model_list_config = {
+        'model': 'my_project.MyModel',
+        'name': 'changelist',
+        'label': 'MyModel Change List',
+        'icon': 'my-model-list-class',
+    }
+
+    #use to create a link of add page
+    my_model_add_config = {
+        'model': 'my_project.MyModel',
+        'name': 'add',
+        'label': 'MyModel Add Item',
+        'icon': 'my-model-add-class',
+    }
+    register_menu_group(position=1, config=my_model_list_config)
+    register_menu_group(position=2, config=my_model_add_config)
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``model``        | (``str``) Model of a app.  (``required``)                    |
++------------------+--------------------------------------------------------------+
+| ``name``         | (``str``) url name. eg. changelist or add.  (``required``)   |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``optional``)         |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the link icon.  (``optional``)   |
++------------------+--------------------------------------------------------------+
+
+Create a menu group
+^^^^^^^^^^^^^^^^^^^
+
+If you want to create a nested menu group then you can use following syntax.
+It creates a dropdown in the menu.
+
+**Syntax:**
+
+.. code-block:: python
+    
+    my_group_items = {
+        1: {'label': 'Link Label', 'url': 'link_url', 'icon': 'my-icon'},
+        2: {
+            'model': 'my_project.MyModel',
+            'name': 'changelist',
+            'label': 'MyModel Change List',
+            'icon': 'my-model-list-class',
+            }
+    }
+    register_menu_group(
+        position=1,
+        config={'label': 'My Label', 'items': my_group_items, 'icon': 'my-icon-class'},
+    )
```

Comment:
same suggestion about simplifying code samples, I'd show one nested data structure, should make it clear that the data is nested.